src/ControlSystem/TimescaleTuner.cpp

```diff
@@ -0,0 +1,126 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ControlSystem/TimescaleTuner.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <cstddef>
+#include <ostream>
+
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/Gsl.hpp"
+
+TimescaleTuner::TimescaleTuner(DataVector initial_timescale,
+                               double max_timescale, double min_timescale,
+                               double max_threshold, double min_threshold,
```

Comment:
Please call these something other than thresholds. There were too many thresholds in SpEC, each doing something different. What do these thresholds do? Can you choose a more descriptive name?

Reply:
Maybe max_threshold -> min_error_to_decrease_timescale and min_threshold -> max_error_to_increase_timescale? Or min_error_tiggering_decrease and max_error_triggering_increase?